tfjs-backend-webgpu/src/kernels/argminmax_webgpu.ts

```diff
@@ -192,4 +196,135 @@ export class ArgMinMaxProgram implements WebGPUProgram {
     `;
     return userCode;
   }
+
+  getUserCodeWgsl(): string {
+    // When this.workGroupSize[0] > 1, each thread reduces Length /
+    // this.workGroupSize[0] values. Thes results are stored in shared memory
+    // and iteratively reduced.
+    const reduceInSharedMemory = this.workGroupSize[0] > 1;
+    const sharedMemorySnippet = `
+      var<workgroup> xBestIndices : array<u32, ${this.workGroupSize[0]}>;
+      var<workgroup> xBestValues : array<f32, ${this.workGroupSize[0]}>;
+    `;
+
+    const sharedMemoryReduceSnippet = `
+      xBestIndices[localId.x] = bestIndex;
+      xBestValues[localId.x] = bestValue;
+
+      var currentSize = WorkGroupSize;
+      for( ;currentSize > 1u; ) {
+        workgroupBarrier();
+
+        for (var w = 0u; w < ${this.reductionFactor}u; w = w + 1u) {
+          let i = localId.x * ${this.reductionFactor}u + w;
+          if (i < currentSize) {
+            let candidateIndex = xBestIndices[i];
+            let candidate = xBestValues[i];
+            if(candidate ${this.op} bestValue && !isNanCustom(candidate)) {
+              bestValue = candidate;
+              bestIndex = candidateIndex;
+            }
+          }
+        }
+
+        xBestIndices[localId.x] = bestIndex;
+        xBestValues[localId.x] = bestValue;
+
+        currentSize = DIV_CEIL(currentSize, ${this.reductionFactor}u);
```

Comment:
I don't see any `break` or `continue` statements, so could this also go in the loop header?

Reply:
You are correct, done, thanks。